test/ERC721HoldingsExecuteCalls.behaviour.js

```diff
@@ -0,0 +1,200 @@
+const Message = artifacts.require('mocks/MessageHelper.sol');
+
+import assertRevert from './helpers/assertRevert';
+const BigNumber = web3.BigNumber;
+
+require('chai')
+  .use(require('chai-as-promised'))
+  .use(require('chai-bignumber')(BigNumber))
+  .should();
+
+export default function shouldExecuteCallsERC721HoldingsToken (accounts) {
+  const firstTokenId = 1;
+  const secondTokenId = 2;
+  const unknownTokenId = 3;
+  
+  const firstAvatarId = 1;
+  const secondAvatarId = 2;
+  const thirdAvatarId = 3;
+  
+  const creator = accounts[0];
+  const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000';
+
+  describe('like an execute calls ERC721HoldingsToken', function () {
+    beforeEach(async function () {
+      this.message = await Message.new({ from: creator });
+
+      await this.avatars.mint(creator, firstAvatarId, { from: creator });
+      await this.avatars.mint(this.message.contract.address, secondAvatarId, { from: creator });
+      await this.avatars.mint(this.token.contract.address, thirdAvatarId, { from: creator });
+
+      await this.token.mint(firstAvatarId, this.avatars.address, firstTokenId, { from: creator });
+      await this.token.mint(firstAvatarId, this.avatars.address, secondTokenId, { from: creator });
+    });
+
+    describe('Test Execute Calls methods', function () { 
+      it('should allow payment through approve', async function () {
+          const extraData = this.message.contract.buyMessage.getData(
```

Comment:
I've used [`encodeCall`](https://github.com/runningbeta/r8-app/blob/master/test/helpers/encodeCall.js) helper in r8-app tests.

It would look something like this:
```
const data = encodeCall('buyMessage', ['bytes32', 'uint256', 'string'], [web3.toHex(123456), 666, 'Transfer Done']);
```

Reply:
You are using the web3 v0.20 API and this [will change with web3 v1.0 release](https://web3js.readthedocs.io/en/1.0/web3-eth-contract.html#methods-mymethod-encodeabi).

Reply:
This is just a FYI comment, no need to change anything for now.